src/main/java/com/blamejared/compat/thermalexpansion/Centrifuge.java

```diff
@@ -64,6 +84,59 @@ protected String getRecipeInfo() {
             return LogHelper.getStackDescription(input);
         }
     }
+
+    private static class AddMob extends BaseAction {
+        private ItemStack[] outputs;
+        private Integer[] chances;
+        private String entityId;
+        private Integer energy;
+        private FluidStack fluid;
+        private Integer xp;
```

Comment:
You don't assign to them anywhere except in construction, please make them final.
Also why use the Wrapper classes?

Reply:
Removed the wrapper; they're not final because the members of the original class I copied from were not final.